wagtail_airtable/management/commands/import_airtable.py

```diff
@@ -359,45 +359,53 @@ def debug_message(message):
                     records_used.append(record['id'])
                     continue
 
+                # Check if we can use the serialized data to create a new model.
+                # If we can, great! If not, fall back to the original mapped_import_fields
+                # If this has to fall back to the original mapped_import_fields: failure
+                # to create a model will be higher than normal.
+                if serialized_data.is_valid():
+                    data_for_new_model = dict(serialized_data.validated_data)
+                else:
+                    data_for_new_model = mapped_import_fields
+                data_for_new_model['airtable_record_id'] = record_id
 
                 # First things first, remove any "pk" or "id" items form the mapped_import_fields
                 # This will let Django and Wagtail handle the PK on its own, as it should.
                 # When the model is saved it'll trigger a push to Airtable and automatically update
                 # the necessary column with the new PK so it's always accurate.
-                try:
-                    del mapped_import_fields['pk']
-                except KeyError:
-                    pass
-                try:
-                    del mapped_import_fields['id']
-                except KeyError:
-                    pass
+                for key in ('pk', 'id',):
+                    try:
+                        del data_for_new_model[key]
+                    except KeyError:
+                        pass
 
                 # If there is no match whatsoever, try to create a new `model` instance.
                 # Note: this may fail if there isn't enough data in the Airtable record.
                 try:
                     debug_message(f"\t\t Attempting to create a new object...")
-                    mapped_import_fields['airtable_record_id'] = record_id
-                    model.objects.create(**mapped_import_fields)
+                    # NOTE: model.objects.create(**data_for_new_model) always throws an
```

Comment:
What's wrong with `model(**data_for_new_model)`? Does that create a similar error?

Reply:
Not sure, I can debug this and report back! That'd be a much nicer way to set all the available fields.

Reply:
Yes, this works perfectly. Will implement this shortly. Thanks Jacob!